invenio_cli/cli/utils.py

```diff
@@ -20,21 +20,19 @@ def run_steps(steps, fail_message, success_message):
     for step in steps:
         click.secho(message=step.message, fg="green")
         result = step.execute()
+
         if result.status_code > 0:
             if result.error:
                 fail_message = fail_message + f"\nErrors: {result.error}"
             if result.output:
                 fail_message = fail_message + f"\nOutput: {result.output}"
 
-            click.secho(fail_message, fg="red")
-            exit(1)
-        elif result.warning:
-            if result.error:
-                fail_message = fail_message + f"\nErrors: {result.error}"
-            if result.output:
-                fail_message = fail_message + f"\nOutput: {result.output}"
+            if result.warning:
+                click.secho(fail_message, fg="yellow")
+            else:
+                click.secho(fail_message, fg="red")
+                exit(1)
```

Comment:
I see... what you say is correct... however, it did not work e.g. for checks on `ImageMagick` [here](https://github.com/inveniosoftware/invenio-cli/blob/master/invenio_cli/commands/requirements.py#L195). I see now, that the correct solution is to add `skippable` to the `FunctionStep`. WDYT?

Since not everybody will use IIIF, the idea was to make the check a warning and allow other checks to be performed.

Reply:
About `I don't know there is clarity on what is the exit code of a command with warning unfortunately`.
I think that we can agree that it is common to expect that a warning will not terminate a program: it will print a message on console but allow the program to continue.

Reply:
Adding skippable to FunctionStep: maybe. We can start with just adding this kind of logic to `check_imagemagick_version` and see from there?

Reply:
That is more or less what I tried to do in the new changes... I guess what you are saying is to merely add `warning=True,` to the `ProcessResponse` of `check_imagemagick_version`... But that is not generic, and we are depending on a "hardcoded" response, while for the `CommandStep` we are depending on `skippable`... What I tried here was to be consistent and depend in the `Step` object, otherwise one will depend on the `Step` and the other directly on the `Response`.